comp/core/config/config.go

```diff
@@ -92,6 +98,22 @@ func newConfig(deps dependencies) (Component, error) {
 		}
 	}
 
+	configRefreshInterval := config.GetDuration("agent_ipc.config_refresh_interval")
+	agentIPCPort := config.GetInt("agent_ipc.port")
+	if flavor.GetFlavor() != flavor.DefaultAgent &&
+		agentIPCPort > 0 &&
+		configRefreshInterval > 0 {
+
+		agentIPCHost := config.GetString("agent_ipc.host")
+		url := &url.URL{
+			Scheme: "https",
+			Host:   net.JoinHostPort(agentIPCHost, strconv.Itoa(agentIPCPort)),
+			Path:   "/config/v1/",
+		}
```

Comment:
Would be nice to have this contained withing the sync logic, so everything is in the same place.

Reply:
Yeah I agree, I initially put that in the function but changed it to ease testing, I'll just use an intermediate function

Reply:
Fixed in [`a1e7b03` (#23211)](https://github.com/DataDog/datadog-agent/pull/23211/commits/a1e7b03ca03b0b07671e15b8cea2233fcb79fa2f)